src/Model/collections.ts

```diff
@@ -152,6 +168,19 @@ Model.prototype.getOrCreateDoc = function(collectionName, id, data) {
   return collection.getOrCreateDoc(id, data);
 };
 
+Model.prototype.getOrDefault = function<S>(subpath: Path, defaultValue: S) {
+  return this.get(subpath) ?? defaultValue as ReadonlyDeep<S>;
+};
+
+Model.prototype.getOrThrow = function<S>(subpath?: Path) {
+  const value = this.get(subpath);
+  if (value === undefined) {
+    const fullpath = [this._at, subpath].filter(Boolean).join('.');
+    throw new Error(`No value at path ${fullpath}`)
```

Comment:
Is there a way we can include the collection name in the error?

Reply:
The full path will already include the collection name - `Model#_at` is the child model's absolute path from the root.